src/traces/sankey/render.js

```diff
@@ -352,8 +376,11 @@ function nodeModel(d, n, i) {
     var visibleThickness = n.dx;
     var visibleLength = Math.max(0.5, n.dy);
 
-    var basicKey = n.label;
-    var key = basicKey + '__' + i;
+    var key = 'node_' + n.pointNumber;
+    // If it's a group, it's mutable and should be unique
+    if(n.group) {
+        key = 'group_' + Math.floor(1e12 * (1 + Math.random()));
```

Comment:
Use 

https://github.com/plotly/plotly.js/blob/7da95e1af39b4b1f98abc2a43deb6b117435931d/src/lib/index.js#L261-L307

instead.

Reply:
I didn't know we had this. I could also reuse it elsewhere in Sankey as can be seen in 1cd206f7fdba75b010650331fc1f7a726ec488df !